ProcessLib/HeatTransportBHE/BHE/BHECommonCoaxial.h

```diff
@@ -0,0 +1,122 @@
+ */
+
+#pragma once
+
+#include <Eigen/Eigen>
+#include "BHECommon.h"
+#include "FlowAndTemperatureControl.h"
+#include "Physics.h"
+#include "PipeConfigurationCoaxial.h"
+#include "ThermalResistancesCoaxial.h"
+#include "ThermoMechanicalFlowProperties.h"
+
+namespace ProcessLib
+{
+namespace HeatTransportBHE
+{
+namespace BHE
+{
+class BHECommonCoaxial : public BHECommon
+{
+public:
+    BHECommonCoaxial(BoreholeGeometry const& borehole,
+                     RefrigerantProperties const& refrigerant,
+                     GroutParameters const& grout,
+                     FlowAndTemperatureControl const& flowAndTemperatureControl,
+                     PipeConfigurationCoaxial const& pipes)
+        : BHECommon{borehole, refrigerant, grout, flowAndTemperatureControl},
+          _pipes(pipes)
+    {
+    }
+
+    static constexpr int number_of_unknowns = 3;
+    static constexpr int number_of_grout_zones = 1;
+
+    double thermalResistance(int const unknown_index) const
+    {
+        return _thermal_resistances[unknown_index];
+    }
+
+    double updateFlowRateAndTemperature(double T_out, double current_time);
+
+    std::array<double, number_of_unknowns> calcThermalResistances(
+        double const Nu_inner_pipe, double const Nu_annulus_pipe);
+
+    std::array<double, number_of_unknowns> pipeHeatCapacities() const;
+
+    static constexpr std::pair<int, int> inflow_outflow_bc_component_ids[] = {
+        {0, 1}};
+
+    std::array<double, number_of_unknowns> pipeHeatConductions() const;
+
+    std::array<Eigen::Vector3d, number_of_unknowns> pipeAdvectionVectors()
+        const;
+    double cross_section_area_inner_pipe, cross_section_area_annulus,
+        cross_section_area_grout;
+
+    std::array<double, number_of_unknowns> crossSectionAreas() const;
+
+    virtual std::array<double, number_of_unknowns> cross_section_areas_coaxial()
+        const = 0;
+
+protected:
+    void updateHeatTransferCoefficients(double const flow_rate)
+    {
+        auto const tm_flow_properties_annulus =
+            calculateThermoMechanicalFlowPropertiesAnnulus(
+                _pipes.inner_pipe,
+                _pipes.outer_pipe,
+                borehole_geometry.length,
+                refrigerant,
+                flow_rate);
+
+        _flow_velocity_annulus = tm_flow_properties_annulus.velocity;
+
+        auto const tm_flow_properties =
+            calculateThermoMechanicalFlowPropertiesPipe(
+                _pipes.inner_pipe,
+                borehole_geometry.length,
+                refrigerant,
+                flow_rate);
+
+        _flow_velocity_inner = tm_flow_properties.velocity;
+
+        _thermal_resistances =
+            calcThermalResistances(tm_flow_properties.nusselt_number,
+                                   tm_flow_properties_annulus.nusselt_number);
+
+        auto const cross_section_area = calculateCrossSectionAreasCoaxial(
+            _pipes.inner_pipe, _pipes.outer_pipe, borehole_geometry.area());
```

Comment:
Do the cross section areas change with the flow_rate? Probably not => no need to calculate them every time.

Reply:
It has been moved.